src/test/scala/com/databricks/spark/avro/AvroSuite.scala

```diff
@@ -297,6 +298,25 @@ class AvroSuite extends FunSuite with BeforeAndAfterAll {
     }
   }
 
+  test("Date field type") {
+    TestUtils.withTempDir { dir =>
+      val schema = StructType(Seq(
+        StructField("float", FloatType, true),
+        StructField("date", DateType, true)
+      ))
+      val rdd = spark.sparkContext.parallelize(Seq(
+        Row(1f, null),
+        Row(2f, new Date(1451980800000L)),
+        Row(3f, new Date(1460098800000L))
+      ))
+      val df = spark.createDataFrame(rdd, schema)
+      df.write.avro(dir.toString)
+      assert(spark.read.avro(dir.toString).count == rdd.count)
+      assert(spark.read.avro(dir.toString).select("date").collect().map(_(0)).toSet ==
+        Array(null, 1451980800000L, 1460098800000L).toSet)
```

Comment:
It's a problem relative to timezone 
[info] - Date field type *** FAILED ***
[info]   Set(1451948400000, 1460066400000, null) did not equal Set(null, 1451980800000, 1460098800000) (AvroSuite.scala:315)

I will investigate